src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -119,37 +251,30 @@ Copyright (c) .NET Foundation. All rights reserved.
     Copy all build outputs, satellites and other necessary files to the publish directory.
     ============================================================
     -->
+  
   <Target Name="CopyFilesToCacheDirectory"
-          DependsOnTargets="_CopyResolvedManagedFiles;
-                            _CopyResolvedUnOptimizedFiles">
-
-    <Message Text="Files were composed in $(PublishDir)"
-             Importance="high"/>
-    <RemoveDir
-        Condition="'$(PreserveComposeWorkingDir)' != 'true'"
-        Directories="$(ComposeWorkingDir)" />
-  </Target>
+          DependsOnTargets="_CopyResolvedOptimizedFiles;
+                            _CopyResolvedUnOptimizedFiles"/>
 
   <!--
     ============================================================
-                                        _CopyResolvedManagedFiles
+                                        _CopyResolvedOptimizedFiles
 
-    Copy _ManagedResolvedFileToPublish items to the publish directory.
+    Copy _CopyResolvedOptimizedFiles items to the publish directory.
     ============================================================
     -->
-  <Target Name="_CopyResolvedManagedFiles"
-          DependsOnTargets="_ComputeResolvedFilesToCacheTypes;
-                            _RunOptimizer"
+  
+  <Target Name="_CopyResolvedOptimizedFiles"
+          DependsOnTargets="_ComputeResolvedFilesToCacheTypes;"
           Inputs="@(_ManagedResolvedFileToPublish)"
           Outputs="@(_ManagedResolvedFileToPublish->'$(PublishDir)%(RecursiveDir)%(Fileame)%(Extension)')">
 
     <Copy SourceFiles = "@(_ManagedResolvedFileToPublish)"
           DestinationFiles="$(PublishDir)%(RecursiveDir)%(Filename)%(Extension)"
+          Condition ="!Exists($(PublishDir)%(RecursiveDir)%(Filename)%(Extension))"
```

Comment:
Do you need this Condition?  What if the file contents changed because I upgraded to a new version of crossgen?